console/DbCommand.cs

```diff
@@ -45,15 +46,17 @@ public abstract class DbCommand : ConsoleCommand {
         protected string DatabaseFilesPath { get; set; }
 
 
-	    protected Database CreateDatabase() {
+	    protected Database CreateDatabase(IList<string> filteredTypes = null) {
+	        filteredTypes = filteredTypes ?? new List<string>();
+
 			if (!string.IsNullOrEmpty(ConnectionString)) {
 				if (!string.IsNullOrEmpty(Server) ||
 			        !string.IsNullOrEmpty(DbName) ||
 					!string.IsNullOrEmpty(User) ||
 					!string.IsNullOrEmpty(Pass)) {
 					throw new ConsoleHelpAsException("You must not provide both a connection string and a server/db/user/password");
 				}
-				return new Database {
+				return new Database(filteredTypes) {
```

Comment:
Should probably follow the established pattern here and use object initializer rather than constructor